src/generators/schema-generator.php

```diff
@@ -198,6 +199,68 @@ protected function add_schema_blocks_graph_pieces( $graph, $context ) {
 		return $graph;
 	}
 
+	/**
+	 * Finalizes the schema graph after all filtering is done.
+	 *
+	 * @param array             $graph   The current schema graph.
+	 * @param Meta_Tags_Context $context The meta tags context.
+	 *
+	 * @return array The schema graph.
+	 */
+	protected function finalize_graph( $graph, $context ) {
+		$graph = $this->remove_empty_breadcrumb( $graph, $context );
+
+		return $graph;
+	}
+
+	/**
+	 * Removes the breadcrumb schema if empty.
+	 *
+	 * @param array             $graph   The current schema graph.
+	 * @param Meta_Tags_Context $context The meta tags context.
+	 *
+	 * @return array The schema graph with empty breadcrumpbs taken out.
+	 */
+	protected function remove_empty_breadcrumb( $graph, $context ) {
+		if ( $this->helpers->current_page->is_home_static_page() || $this->helpers->current_page->is_home_posts_page() ) {
+			return $graph;
+		}
+
+		// Remove the breadcrumb piece, if it's empty.
+		$index_to_remove = 0;
+		foreach ( $graph as $key => $piece ) {
+			if ( \in_array( 'BreadcrumbList', $this->get_type_from_piece( $piece ), true ) ) {
+				if ( isset( $piece['itemListElement'] ) && is_array( $piece['itemListElement'] ) && count( $piece['itemListElement'] ) === 1 ) {
+					$index_to_remove = $key;
+					break;
+				}
+			}
+		}
+
+		// If the breadcrumb piece has been removed, we should remove its reference from the WebPage node.
+		if ( $index_to_remove !== 0 ) {
+			\array_splice( $graph, $index_to_remove, 1 );
+
+			// Get the type of the WebPage node.
+			$webpage_type = $context->schema_page_type;
+			if ( ! is_array( $webpage_type ) ) {
+				$webpage_types = [ $webpage_type ];
+			}
+			else {
+				$webpage_types = $webpage_type;
+			}
+
+			foreach ( $graph as $key => $piece ) {
+				if ( ! empty( \array_intersect( $webpage_types, $this->get_type_from_piece( $piece ) ) ) && isset( $piece['breadcrumb'] ) ) {
```

Comment:
Why did you choose to do an ! empty intersect here instead of another in_array as above. I think that would be possible right, because you are checking are the types I am looking for in the type array from the graph piece?

Reply:
Because contrary to the above in_array, here the needle is an array, so in_array doesn't work